include/oneapi/dpl/internal/dynamic_selection_impl/dynamic_load_policy.h

```diff
@@ -89,79 +91,57 @@ struct dynamic_load_policy
             resource_->load_.fetch_sub(1);
         }
     };
+    using selection_type = dl_selection_handle_t<dynamic_load_policy<ResourceType, ResourceAdapter, Backend>>;
 
-    std::shared_ptr<backend_t> backend_;
-
-    using selection_type = dl_selection_handle_t<dynamic_load_policy<Backend>>;
-
-    struct state_t
+    struct selector_t
     {
         resource_container_t resources_;
         std::mutex m_;
     };
 
-    std::shared_ptr<state_t> state_;
+    std::shared_ptr<selector_t> selector_;
```

Comment:
As part of this refactoring, can `selector_` be made a unique pointer?

Reply:
This is to allow policies to be lightweight and to make them able to be copied around freely without adding to the footprint substantially (shallow copy of the shared pointer which contains the non-trivial footprint of the policy).  Unique pointer would prevent policies from being copied, and defeat the purpose.